### PR TITLE
[RViz] Check valid interactive marker pointer in Motion Planning Display when changing fixed frame

### PR DESCRIPTION
**Pull request by sea-bass**
### Description

While going through the [MoveIt QuickStart in RViz Tutorial](https://moveit.picknik.ai/humble/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html), I was getting RViz crashing when I changed the fixed frame as shown here:

image

I chased this down to `MotionPlanningDisplay::fixedFrameChanged()` in `moveit2/moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp` and found that the issue was resolved if I checked that the pointers `frame_->scene_marker_` were non-null.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [x] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [x] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [x] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp [hunk 1]
```diff
@@ -1422,8 +1422,9 @@ void MotionPlanningDisplay::fixedFrameChanged()
   if (int_marker_display_)
     int_marker_display_->setFixedFrame(fixed_frame_);
   // When the fixed frame changes we need to tell RViz to update the rendered interactive marker display
-  frame_->scene_marker_->requestPoseUpdate(frame_->scene_marker_->getPosition(),
-                                           frame_->scene_marker_->getOrientation());
+  if (frame_ && frame_->scene_marker_)
+    frame_->scene_marker_->requestPoseUpdate(frame_->scene_marker_->getPosition(),
+                                             frame_->scene_marker_->getOrientation());
```
Comment (tylerjw) [inline moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp:1427]: Should we log something when this is false? Ideally, we'd send the error some other way... I generally don't like reporting errors in logs if we can help it but it is definitely better to not dereference an invalid pointer than what is here.

Comment (sea-bass) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp:1427]: I took a deeper look, and `frame_` is valid but `frame->scene_marker_` is not.

This is totally fine given how `MotionPlanningFrame::createSceneInteractiveMarker()` in `moveit2/moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_objects.cpp` doesn't always instantiate a scene marker -- in fact, this code path only makes a marker if we have a collision object selected in the scene, as in here:

image

Also, only a few lines above we have a similar valid pointer check for the `int_marker_display_` variable (which is the interactive marker on the move group's end effector frame) and we don't log anything on that pointer being null.

Comment (tylerjw) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp:1427]: Thank you for the detailed response.

**Review by tylerjw — APPROVED** [review]

**Review by vatanaksoytezer — APPROVED** [review]

**Comment by MarcoMagriDev** [comment]
I have encountered the same issue using `humble` branch and I noticed that this PR has not been backported to that branch [see here](https://github.com/ros-planning/moveit2/blob/b679dd870031ad86f8fb5c58df42f7649ab64fc2/moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp#L1424).

Could you please backport this PR to `humble`?

**Comment by sea-bass** [comment]
@MarcoMagriDev good call -- opened up https://github.com/ros-planning/moveit2/pull/2366

**Pull request merged**